strongback-log-processing/src/org/frc4931/utils/LogDecoder.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * FRC 4931 (http://www.evilletech.com)
+ *
+ * Open source software. Licensed under the FIRST BSD license file in the
+ * root directory of this project's Git repository.
+ */
+package org.frc4931.utils;
+import java.io.BufferedInputStream;
+import java.io.BufferedWriter;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Utility to process data logs generated by strongback into human readable csv format. The path to the input 
+ * file should be specified as the first command line argument. The path to the output is optional, if it is not
+ * specified, the output will be saved in the current directory, with the same filename as the input, and a .csv
+ * extension. <br>
+ * {@code Usage: LogDecoder <input_path> [output_path]}
+ * 
+ * <h1>Log file format:</h1>
+ * <p>
+ * The first three bytes are the ASCII log ({@code 6C 6F 67}). The next byte is the number of elements that have 
+ * been logged {@code n}. The next {@code n} bytes are the number of bytes in each data point, followed by 
+ * {@code n} repetitions of the length of the name of each data point and the name itself in ASCII bytes.
+ * <p>
+ * Now the data is recorded with respect to the number of bytes in each element. 
+ * When Logger is stopped it will finish writing the line it is on and then the terminator: {@code FF FF FF FF}.
+ * 
+ * <h1>Example log file:</h1>
+ * <pre> {@code [l o g] 
```

Comment:
This is a binary file, right? Might want to preface this `pre` block with a sentence that says the sample demonstrates the layout of the binary file, and that line feeds don't appear in the file and are included in the example only for clarity.